File: OP_PROJECT_PYTHON/src/game/Plants/SosnowskyHogweed.py
from src.game.Plant import Plant
from src.game.Position import Position


class SosnowskyHogweed(Plant):
    SOSNOWSKY_HOGWEED_STRENGTH = 10
    SOSNOWSKY_HOGWEED_TYPE = "Sosnowsky"

    def __init__(self, position, world):
        super().__init__(SosnowskyHogweed.SOSNOWSKY_HOGWEED_TYPE, SosnowskyHogweed.SOSNOWSKY_HOGWEED_STRENGTH, position, world)

    def action(self):
        current_position = self.position

        for dy in range(-1, 2):
            for dx in range(-1, 2):
                if dx == 0 and dy == 0:
                    continue
                neighbor_position = Position(current_position.get_y() + dy, current_position.get_x() + dx)

                if self.world.is_position_within_bounds(neighbor_position):
                    neighbor_organism = self.world.get_organism_at_position(neighbor_position)

                    if neighbor_organism is not None and neighbor_organism.get_group() == "ANIMAL" and neighbor_organism.get_type() != "Cyber Sheep":
                        self.world.remove_organism_at_position(neighbor_position)
                        info = "Sosnowsky Hogweed killed: " + neighbor_organism.get_type()
                        self.world.display_information(info)
                        self.world.remove_organism(neighbor_organism)

File: OP_PROJECT_JAVA/src/game/Plants/SosnowskyHogweed.java
package game.Plants;

import game.Organism;
import game.OrganismGroup;
import game.Plant;
import game.Position;
import game.World;

public class SosnowskyHogweed extends Plant {

    private static final int SOSNOWSKY_HOGWEED_STRENGTH = 10;
    private static final String SOSNOWSKY_HOGWEED_TYPE = "Sosnowsky";


    public SosnowskyHogweed(Position position, World world) {
        super(SOSNOWSKY_HOGWEED_TYPE, SOSNOWSKY_HOGWEED_STRENGTH, position, world);
    }

    @Override
    public void action() {
        Position currentPosition = position;


        for (int dy = -1; dy <= 1; ++dy) {
            for (int dx = -1; dx <= 1; ++dx) {
                if (dx == 0 && dy == 0) {
                    continue;
                }
                Position neighborPosition = new Position(currentPosition.getY() + dy, currentPosition.getX() + dx);

                if (world.isPositionWithinBounds(neighborPosition)) {
                    Organism neighborOrganism = world.getOrganismAtPosition(neighborPosition);

                    if (neighborOrganism != null && neighborOrganism.getGroup() == OrganismGroup.ANIMAL) {
                        world.removeOrganismAtPosition(neighborPosition);
                        String info = "Sosnowsky Hogweed killed: " + neighborOrganism.getType();
                        world.displayInformation(info);
                        world.removeOrganism(neighborOrganism);
                    }
                }
            }
        }
    }
}
